pkg/controller/kfservice/reconcilers/knative/service_reconciler.go

```diff
@@ -159,13 +159,13 @@ func (r *ServiceReconciler) reconcileService(kfsvc *v1alpha2.KFService, desired
 	}
 
 	// Reconcile differences and update
-	diff, err := kmp.SafeDiff(desired.Spec, existing.Spec)
+	diff, err := kmp.SafeDiff(desired.Spec.ConfigurationSpec, existing.Spec.ConfigurationSpec)
```

Comment:
What's going on with this diff here? Should we also reduce the scope of semanticEquals to what's being diffed?